File: src/main/java/com/indiepost/dto/post/PostUserInteraction.java
package com.indiepost.dto.post;

import java.time.Instant;

public class PostUserInteraction {

    private final Long postId;

    private Instant lastRead;

    private Instant bookmarked;

    public PostUserInteraction(Long postId) {
        this.postId = postId;
    }

    public Long getPostId() {
        return postId;
    }

    public Instant getLastRead() {
        return lastRead;
    }

    public void setLastRead(Instant lastRead) {
        this.lastRead = lastRead;
    }

    public Instant getBookmarked() {
        return bookmarked;
    }

    public void setBookmarked(Instant bookmarked) {
        this.bookmarked = bookmarked;
    }
}

File: src/main/kotlin/com/indiepost/dto/post/PostUserInteraction.kt
package com.indiepost.dto.post

import java.time.Instant

data class PostUserInteraction(
        var postId: Long? = null,

        var lastRead: Instant? = null,

        var bookmarked: Instant? = null
) {
    constructor(postId: Long) : this() {
        this.postId = postId
    }
}

